In [4]:
# importing library

import pandas as pd

In [15]:
# loading data

data = pd.read_csv('C:\\Users\\Admin\\Internshala\\mode_dataset.csv')

In [17]:
# printing first rows

data.head()

Student  Overall Marks Gender    Subject
0    Anish             97   Male    English
1  Rishabh             83   Male    English
2     Ajay             78   Male    Physics
3  Abhinav             86   Male  Chemistry
4   Anurag             91   Male    English

In [18]:
# printing bottom rows

data.tail()

Student  Overall Marks  Gender    Subject
95  Aaditya             77    Male  Chemistry
96    vivek             55    Male    Physics
97    Tisha              7  Female    English
98   Raghav             73    Male    English
99   Ishita             82  Female  Chemistry

In [19]:
# calculating the mean

mean_data = data['Overall Marks'].mean()

In [20]:
print(mean_data)

68.21


In [21]:
# calculate the mode and print the mode

mode_data = data['Overall Marks'].mode()

In [22]:
print(mode_data)

0    88
dtype: int64


In [36]:
# Added a value 1000 at the bottom of a file and save it as mean_robust.csv

data_new = pd.read_csv('C:\\Users\\Admin\\Internshala\\mean_robust.csv')

In [37]:
# print last rows

data_new.tail()

Student  Overall Marks  Gender    Subject
95  Aaditya             77    Male  Chemistry
96    vivek             55    Male    Physics
97    Tisha              7  Female    English
98   Raghav             73    Male    English
99   Ishita          10000  Female  Chemistry

In [38]:
mean_data_new = data_new['Overall Marks'].mean()

In [39]:
print(mean_data_new)

167.39


NOTE: The total marks that students can get 100, and the mean is coming as 167.39
As we can see it is not a complete robust method at all.
And we were calculating "mode", since it is coming as 88.
s we can see there is a quite difference there between the robustness of these two metrix.